src/js/contentscripts/socialwidgets.js

```diff
@@ -443,8 +443,29 @@ function createReplacementWidget(widget, elToReplace, activationFn) {
 
   let textDiv = document.createElement('div');
   textDiv.style = styleAttrs.join(" !important;") + " !important";
-  textDiv.appendChild(document.createTextNode(
-    TRANSLATIONS.widget_placeholder_pb_has_replaced.replace("XXX", name)));
+  // add link to replaced widget text if it has a src
+  if (elToReplace.getAttribute("src")) {
+    let widget_link = document.createElement("a");
+    widget_link.rel = "noreferrer";
+    widget_link.target = "_blank";
+    widget_link.href = elToReplace.src;
+    widget_link.innerHTML = name + ' widget';
+
+    let widget_link_styles = [
+      "color: #ec9329",
+      "margin-left: 3px",
+      "text-decoration: none"
+    ];
+
+    widget_link.style = widget_link_styles.join(" !important;") + " !important";
+
+    textDiv.innerHTML = TRANSLATIONS.widget_placeholder_pb_has_replaced.replace("XXX widget", '');
```

Comment:
This doesn't work in non-English locales.

Reply:
`innerHTML` is dangerous (could lead to XSS) and may cause problems with extension store reviewers.

Reply:
We should include the word "this" in the link probably. As in, "this Vimeo widget".

Reply:
To resolve the above, we could add new placeholders to the `widget_placeholder_pb_has_replaced` locale string that mark the start and the end of the anchor tag.

We would insert temporary markers when fetching localized widget strings in `webrequest.js`, similarly to how we insert `"XXX"` now.

Then we would either remove those markers when we don't have a link to build, or we would use them to construct the link. This should work in both LTR and RTL locales.

Reply:
To avoid using `innerHTML`, we could use the temporary markers to split the string up into three chunks: one for text before the link (could be empty), one for text inside the link, and one for text following the link (could be empty). We would then construct a text node (if not empty), a link element and another text node (if not empty) from those chunks.

Reply:
Agreed on removing `innerHTML` ! That's been taken care of in the latest commit.

Also in the latest commit I started moving towards breaking apart the replaced text divs into three chunks, with placeholders used in the i18n string. To be honest, I don't quite understand the advantage of that approach. Isn't it easier to manipulate the innerText of the anchor tag directly in the socialwidgets script?

Reply:
How do you know what the link text is? You have to extract it from the localized string.